File: exercises/leetcode/algorithms/121-best-time-to-buy-and-sell-stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min_price = float('inf')
        max_profit = 0

        for price in prices:
            min_price = min(price, min_price)
            max_profit = max(max_profit, price - min_price)

        return max_profit

File: exercises/leetcode/algorithms/121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
   let minPrice = Number.MAX_SAFE_INTEGER
   let maxProfit = 0

   for (const price of prices) {
      minPrice = Math.min(price, minPrice)
      maxProfit = Math.max(maxProfit, price - minPrice)
   }

   return maxProfit
}
